src/statsmodel.jl

```diff
@@ -45,12 +45,23 @@ struct DataFrameRegressionModel{M,T} <: RegressionModel
     mm::ModelMatrix{T}
 end
 
+"""
+    supports_intercept(::Any)
```

Comment:
Would better be `::Type` IIUC, right?

Also wrap lines at 92 chars, and use imperative ("define") rather than third person. Better say "Return `true` by default.", and "model type" rather than "model". Also "will be fitted".